src/main/java/io/cdap/plugin/google/common/GoogleDriveBaseConfig.java

```diff
@@ -67,49 +102,80 @@
   public void validate(FailureCollector collector) {
     IdUtils.validateReferenceName(referenceName, collector);
 
-    if (validateAccountFilePath(collector)) {
-      try {
-        GoogleDriveClient client = getDriveClient();
-
-        // validate auth
-        validateCredentials(collector, client);
+    if (validateAuthType(collector)) {
+      AuthType authType = getAuthType();
+      boolean propertiesAreValid = false;
+      switch (authType) {
```

Comment:
I think we should handle a default here: with an exception.

Reply:
done